ktor-client/ktor-client-core/common/src/io/ktor/client/request/HttpRequest.kt

```diff
@@ -217,6 +217,39 @@ public class HttpResponseData constructor(
  */
 public fun HttpRequestBuilder.headers(block: HeadersBuilder.() -> Unit): HeadersBuilder = headers.apply(block)
 
+/**
+ * Executes a [block] that configures the [CookiesBuilder] associated to this request.
+ */
+public fun HttpRequestBuilder.cookies(block: CookiesBuilder.() -> Unit): HeadersBuilder =
+    CookiesBuilder(headers).apply(block).headersBuilder
+
+public class CookiesBuilder(internal val headersBuilder: HeadersBuilder) {
```

Comment:
Is this the right place for this class? Isn't it better to have it in `io.ktor.client.features.cookies` package

Reply:
I would keep it in the same package but extract in the separate file

Reply:
Good point, thanks, will move.